app/kumactl/cmd/apply/apply.go

```diff
@@ -83,39 +83,41 @@ func NewApplyCmd(pctx *kumactl_cmd.RootContext) *cobra.Command {
 					}
 				}
 			}
+			resources := strings.Split(string(b), "---")
+			for _, resource := range resources {
```

Comment:
please add a check
```
if len(resource) == 0 {
  continue
}
```
the reason is that you can start the multiresource YAML with `---`